### PR TITLE
feat: add createSignal and createComputed helper functions

### PR DESCRIPTION
**Pull request by eneajaho**

### CODE REVIEW

File: libs/ngxtension/create-signal/src/create-signal.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			get() {
+				return sig();
+			},
+			set(value: T) {
+				sig.set(value);
+			},
+		},
+	});
+
+	return sig as WritableSignal<T> & { value: T };
```
Comment (nartc) [inline libs/ngxtension/create-signal/src/create-signal.ts:47]: nit/suggestion: let's change this to `defineProperty` instead

```suggestion
	return Object.defineProperty(sig, 'value', {
    get: sig.asReadonly(),
    set: sig.set.bind(sig),
	}) as WritableSignal<T> & { value: T };
```

**Review by nartc — CHANGES_REQUESTED** [review]

File: libs/ngxtension/create-signal/src/create-signal.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+	Object.defineProperties(sig, {
+		value: {
+			get() {
+				return sig();
+			},
+		},
+	});
+
+	return sig as Signal<T> & { value: T };
```
Comment (nartc) [inline libs/ngxtension/create-signal/src/create-signal.ts:55]: nit: c'mon boi! why not change this here as well xD

**Review by nartc — CHANGES_REQUESTED** [review]

File: libs/ngxtension/create-signal/src/create-signal.ts (hunk 2 above)
Comment (eneajaho) [reply libs/ngxtension/create-signal/src/create-signal.ts:55]: There's no readonly for computed?

Comment (nartc) [reply libs/ngxtension/create-signal/src/create-signal.ts:55]: yep but `defineProperty` and also you can pass a reference to `sig` to `get` instead:

```ts
return Object.defineProperty(sig, 'value', { get: sig })
```

**Review by nartc — APPROVED** [review]

**Pull request merged**